classes/SectionView/SectionView.php

```diff
@@ -21,23 +22,116 @@ public function __construct() {
     }
 
     public function display() {
-        global $PAGE;
         if ($year = get_http_var('y')) {
-            $this->display_year($year);
+            $data = $this->display_year($year);
+            $data['year'] = $year;
+            $data['template'] = 'section/year';
+            $data = $this->addCommonData($data);
         } elseif (($date = get_http_var('d')) && ($column = get_http_var('c'))) {
-            $this->display_column($date, $column);
+            $data = $this->display_column($date, $column);
+            $data = $this->addCommonData($data);
         } elseif ($date = get_http_var('d')) {
-            $this->display_day($date);
+            $data = $this->display_day($date);
+            if ( !isset($data['template']) ) {
+                $data['template'] = 'section/day';
+            }
+            $data = $this->addCommonData($data);
         } elseif (get_http_var('id')) {
             $data = $this->display_section_or_speech();
-            return $data;
         } else {
-            $skip_end = $this->display_front();
-            if ( $skip_end ) {
-                return;
+            $data = $this->display_front();
+            if ( !isset($data['template']) ) {
+                $data['template'] = 'section/recent';
             }
+            $data['search_sections'] = $this->getSearchSections();
+            $data = $this->addCommonData($data);
+        }
+
+        return $data;
+    }
+
+    protected function addCommonData($data) {
+        global $DATA, $this_page;
+
+        $common = new \MySociety\TheyWorkForYou\Common;
+        $data['urls'] = $this->getURLs($data);
+        $data['popular_searches'] = $common->getPopularSearches();
+        $data['recess_major'] = $this->getRecessMajor($data);
+
+        $nextprev = $DATA->page_metadata($this_page, 'nextprev');
```

Comment:
This is duplicated in getURLs (which puts next/prev URLs only in $urls) - are both needed?